File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/TaskRunOutput.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Output of a task run.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=TaskRunOutput.Builder.class)

public final class TaskRunOutput  {
    @Deprecated
    @java.beans.ConstructorProperties({"key", "taskType", "isTruncated", "errorTrace", "data", "outputParameters", "version"})
    public TaskRunOutput(String key, TaskType taskType, Boolean isTruncated, String errorTrace, java.util.List<RunOutputData> data, java.util.List<OutputParameter> outputParameters, Integer version) {
        super();
        this.key = key;
        this.taskType = taskType;
        this.isTruncated = isTruncated;
        this.errorTrace = errorTrace;
        this.data = data;
        this.outputParameters = outputParameters;
        this.version = version;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * A unique identifier for the output.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("key")
private String key;

        /**
         * A unique identifier for the output.
         * @param key the value to set
         * @return this builder
         **/
        

public Builder key(String key) {
    this.key = key;
    return this;
}
            /**
     * The type of the task.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("taskType")
private TaskType taskType;

        /**
         * The type of the task.
         * @param taskType the value to set
         * @return this builder
         **/
        

public Builder taskType(TaskType taskType) {
    this.taskType = taskType;
    return this;
}
            /**
     * True if logs are truncated.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("isTruncated")
private Boolean isTruncated;

        /**
         * True if logs are truncated.
         * @param isTruncated the value to set
         * @return this builder
         **/
        

public Builder isTruncated(Boolean isTruncated) {
    this.isTruncated = isTruncated;
    return this;
}
            /**
     * If there was an error executing the run, this field contains any available stack traces.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("errorTrace")
private String errorTrace;

        /**
         * If there was an error executing the run, this field contains any available stack traces.
         * @param errorTrace the value to set
         * @return this builder
         **/
        

public Builder errorTrace(String errorTrace) {
    this.errorTrace = errorTrace;
    return this;
}
            /**
     * Array of output objects.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("data")
private java.util.List<RunOutputData> data;

        /**
         * Array of output objects.
         * @param data the value to set
         * @return this builder
         **/
        

public Builder data(java.util.List<RunOutputData> data) {
    this.data = data;
    return this;
}
            /**
     * List of output parameters with name and values.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("outputParameters")
private java.util.List<OutputParameter> outputParameters;

        /**
         * List of output parameters with name and values.
         * @param outputParameters the value to set
         * @return this builder
         **/
        

public Builder outputParameters(java.util.List<OutputParameter> outputParameters) {
    this.outputParameters = outputParameters;
    return this;
}
            /**
     * Current version of job run object in repository.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("version")
private Integer version;

        /**
         * Current version of job run object in repository.
         * @param version the value to set
         * @return this builder
         **/
        

public Builder version(Integer version) {
    this.version = version;
    return this;
}


        public TaskRunOutput build() {
            TaskRunOutput model = new TaskRunOutput(this.key
                , this.taskType
                , this.isTruncated
                , this.errorTrace
                , this.data
                , this.outputParameters
                , this.version);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(TaskRunOutput model) {
                this.key(model.getKey());
    this.taskType(model.getTaskType());
    this.isTruncated(model.getIsTruncated());
    this.errorTrace(model.getErrorTrace());
    this.data(model.getData());
    this.outputParameters(model.getOutputParameters());
    this.version(model.getVersion());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * A unique identifier for the output.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("key")
    private final String key;

        /**
     * A unique identifier for the output.
     * @return the value
     **/
    
    public String getKey() {
        return key;
    }

    /**
     * The type of the task.
     **/
    public enum TaskType implements com.oracle.bmc.http.internal.BmcEnum {
        NotebookTask("NOTEBOOK_TASK"),
        PythonTask("PYTHON_TASK"),
        SparkSubmitTask("SPARK_SUBMIT_TASK"),
        IfElseTask("IF_ELSE_TASK"),
        JobTask("JOB_TASK"),
        JarTask("JAR_TASK"),
        

        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        UnknownEnumValue(null);

        private static final org.slf4j.Logger LOG = org.slf4j.LoggerFactory.getLogger(TaskType.class);

        private final String value;
        private static java.util.Map<String, TaskType> map;

        static {
            map = new java.util.HashMap<>();
            for (TaskType v : TaskType.values()) {
                if (v != UnknownEnumValue) {
                    map.put(v.getValue(), v);
                }
            }
        }

        TaskType(String value) {
            this.value = value;
        }

        @com.fasterxml.jackson.annotation.JsonValue
        public String getValue() {
            return value;
        }

        @com.fasterxml.jackson.annotation.JsonCreator
        public static TaskType create(String key) {
            if (map.containsKey(key)) {
                return map.get(key);
            }
            LOG.warn("Received unknown value '{}' for enum 'TaskType', returning UnknownEnumValue", key);
            return UnknownEnumValue;
        }
    };
        /**
     * The type of the task.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("taskType")
    private final TaskType taskType;

        /**
     * The type of the task.
     * @return the value
     **/
    
    public TaskType getTaskType() {
        return taskType;
    }


        /**
     * True if logs are truncated.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("isTruncated")
    private final Boolean isTruncated;

        /**
     * True if logs are truncated.
     * @return the value
     **/
    
    public Boolean getIsTruncated() {
        return isTruncated;
    }


        /**
     * If there was an error executing the run, this field contains any available stack traces.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("errorTrace")
    private final String errorTrace;

        /**
     * If there was an error executing the run, this field contains any available stack traces.
     * @return the value
     **/
    
    public String getErrorTrace() {
        return errorTrace;
    }


        /**
     * Array of output objects.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("data")
    private final java.util.List<RunOutputData> data;

        /**
     * Array of output objects.
     * @return the value
     **/
    
    public java.util.List<RunOutputData> getData() {
        return data;
    }


        /**
     * List of output parameters with name and values.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("outputParameters")
    private final java.util.List<OutputParameter> outputParameters;

        /**
     * List of output parameters with name and values.
     * @return the value
     **/
    
    public java.util.List<OutputParameter> getOutputParameters() {
        return outputParameters;
    }


        /**
     * Current version of job run object in repository.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("version")
    private final Integer version;

        /**
     * Current version of job run object in repository.
     * @return the value
     **/
    
    public Integer getVersion() {
        return version;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("TaskRunOutput(");
        sb.append("key=").append(String.valueOf(this.key));
        sb.append(", taskType=").append(String.valueOf(this.taskType));
        sb.append(", isTruncated=").append(String.valueOf(this.isTruncated));
        sb.append(", errorTrace=").append(String.valueOf(this.errorTrace));
        sb.append(", data=").append(String.valueOf(this.data));
        sb.append(", outputParameters=").append(String.valueOf(this.outputParameters));
        sb.append(", version=").append(String.valueOf(this.version));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof TaskRunOutput)) {
            return false;
        }

        TaskRunOutput other = (TaskRunOutput) o;
        return java.util.Objects.equals(this.key, other.key) &&
            java.util.Objects.equals(this.taskType, other.taskType) &&
            java.util.Objects.equals(this.isTruncated, other.isTruncated) &&
            java.util.Objects.equals(this.errorTrace, other.errorTrace) &&
            java.util.Objects.equals(this.data, other.data) &&
            java.util.Objects.equals(this.outputParameters, other.outputParameters) &&
            java.util.Objects.equals(this.version, other.version);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.key == null ? 43 : this.key.hashCode());
        result = (result * PRIME) + (this.taskType == null ? 43 : this.taskType.hashCode());
        result = (result * PRIME) + (this.isTruncated == null ? 43 : this.isTruncated.hashCode());
        result = (result * PRIME) + (this.errorTrace == null ? 43 : this.errorTrace.hashCode());
        result = (result * PRIME) + (this.data == null ? 43 : this.data.hashCode());
        result = (result * PRIME) + (this.outputParameters == null ? 43 : this.outputParameters.hashCode());
        result = (result * PRIME) + (this.version == null ? 43 : this.version.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/task_run_output.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class TaskRunOutput(object):
    """
    Output of a task run.
    """

    #: A constant which can be used with the task_type property of a TaskRunOutput.
    #: This constant has a value of "NOTEBOOK_TASK"
    TASK_TYPE_NOTEBOOK_TASK = "NOTEBOOK_TASK"

    #: A constant which can be used with the task_type property of a TaskRunOutput.
    #: This constant has a value of "PYTHON_TASK"
    TASK_TYPE_PYTHON_TASK = "PYTHON_TASK"

    #: A constant which can be used with the task_type property of a TaskRunOutput.
    #: This constant has a value of "SPARK_SUBMIT_TASK"
    TASK_TYPE_SPARK_SUBMIT_TASK = "SPARK_SUBMIT_TASK"

    #: A constant which can be used with the task_type property of a TaskRunOutput.
    #: This constant has a value of "IF_ELSE_TASK"
    TASK_TYPE_IF_ELSE_TASK = "IF_ELSE_TASK"

    #: A constant which can be used with the task_type property of a TaskRunOutput.
    #: This constant has a value of "JOB_TASK"
    TASK_TYPE_JOB_TASK = "JOB_TASK"

    #: A constant which can be used with the task_type property of a TaskRunOutput.
    #: This constant has a value of "JAR_TASK"
    TASK_TYPE_JAR_TASK = "JAR_TASK"

    def __init__(self, **kwargs):
        """
        Initializes a new TaskRunOutput object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param key:
            The value to assign to the key property of this TaskRunOutput.
        :type key: str

        :param task_type:
            The value to assign to the task_type property of this TaskRunOutput.
            Allowed values for this property are: "NOTEBOOK_TASK", "PYTHON_TASK", "SPARK_SUBMIT_TASK", "IF_ELSE_TASK", "JOB_TASK", "JAR_TASK", 'UNKNOWN_ENUM_VALUE'.
            Any unrecognized values returned by a service will be mapped to 'UNKNOWN_ENUM_VALUE'.
        :type task_type: str

        :param is_truncated:
            The value to assign to the is_truncated property of this TaskRunOutput.
        :type is_truncated: bool

        :param error_trace:
            The value to assign to the error_trace property of this TaskRunOutput.
        :type error_trace: str

        :param data:
            The value to assign to the data property of this TaskRunOutput.
        :type data: list[oci.aidataplatform_dp.models.RunOutputData]

        :param output_parameters:
            The value to assign to the output_parameters property of this TaskRunOutput.
        :type output_parameters: list[oci.aidataplatform_dp.models.OutputParameter]

        :param version:
            The value to assign to the version property of this TaskRunOutput.
        :type version: int

        """
        self.swagger_types = {
            'key': 'str',
            'task_type': 'str',
            'is_truncated': 'bool',
            'error_trace': 'str',
            'data': 'list[RunOutputData]',
            'output_parameters': 'list[OutputParameter]',
            'version': 'int'
        }

        self.attribute_map = {
            'key': 'key',
            'task_type': 'taskType',
            'is_truncated': 'isTruncated',
            'error_trace': 'errorTrace',
            'data': 'data',
            'output_parameters': 'outputParameters',
            'version': 'version'
        }

        self._key = None
        self._task_type = None
        self._is_truncated = None
        self._error_trace = None
        self._data = None
        self._output_parameters = None
        self._version = None

    @property
    def key(self):
        """
        Gets the key of this TaskRunOutput.
        A unique identifier for the output.


        :return: The key of this TaskRunOutput.
        :rtype: str
        """
        return self._key

    @key.setter
    def key(self, key):
        """
        Sets the key of this TaskRunOutput.
        A unique identifier for the output.


        :param key: The key of this TaskRunOutput.
        :type: str
        """
        self._key = key

    @property
    def task_type(self):
        """
        Gets the task_type of this TaskRunOutput.
        The type of the task.

        Allowed values for this property are: "NOTEBOOK_TASK", "PYTHON_TASK", "SPARK_SUBMIT_TASK", "IF_ELSE_TASK", "JOB_TASK", "JAR_TASK", 'UNKNOWN_ENUM_VALUE'.
        Any unrecognized values returned by a service will be mapped to 'UNKNOWN_ENUM_VALUE'.


        :return: The task_type of this TaskRunOutput.
        :rtype: str
        """
        return self._task_type

    @task_type.setter
    def task_type(self, task_type):
        """
        Sets the task_type of this TaskRunOutput.
        The type of the task.


        :param task_type: The task_type of this TaskRunOutput.
        :type: str
        """
        allowed_values = ["NOTEBOOK_TASK", "PYTHON_TASK", "SPARK_SUBMIT_TASK", "IF_ELSE_TASK", "JOB_TASK", "JAR_TASK"]
        if not value_allowed_none_or_none_sentinel(task_type, allowed_values):
            task_type = 'UNKNOWN_ENUM_VALUE'
        self._task_type = task_type

    @property
    def is_truncated(self):
        """
        Gets the is_truncated of this TaskRunOutput.
        True if logs are truncated.


        :return: The is_truncated of this TaskRunOutput.
        :rtype: bool
        """
        return self._is_truncated

    @is_truncated.setter
    def is_truncated(self, is_truncated):
        """
        Sets the is_truncated of this TaskRunOutput.
        True if logs are truncated.


        :param is_truncated: The is_truncated of this TaskRunOutput.
        :type: bool
        """
        self._is_truncated = is_truncated

    @property
    def error_trace(self):
        """
        Gets the error_trace of this TaskRunOutput.
        If there was an error executing the run, this field contains any available stack traces.


        :return: The error_trace of this TaskRunOutput.
        :rtype: str
        """
        return self._error_trace

    @error_trace.setter
    def error_trace(self, error_trace):
        """
        Sets the error_trace of this TaskRunOutput.
        If there was an error executing the run, this field contains any available stack traces.


        :param error_trace: The error_trace of this TaskRunOutput.
        :type: str
        """
        self._error_trace = error_trace

    @property
    def data(self):
        """
        **[Required]** Gets the data of this TaskRunOutput.
        Array of output objects.


        :return: The data of this TaskRunOutput.
        :rtype: list[oci.aidataplatform_dp.models.RunOutputData]
        """
        return self._data

    @data.setter
    def data(self, data):
        """
        Sets the data of this TaskRunOutput.
        Array of output objects.


        :param data: The data of this TaskRunOutput.
        :type: list[oci.aidataplatform_dp.models.RunOutputData]
        """
        self._data = data

    @property
    def output_parameters(self):
        """
        Gets the output_parameters of this TaskRunOutput.
        List of output parameters with name and values.


        :return: The output_parameters of this TaskRunOutput.
        :rtype: list[oci.aidataplatform_dp.models.OutputParameter]
        """
        return self._output_parameters

    @output_parameters.setter
    def output_parameters(self, output_parameters):
        """
        Sets the output_parameters of this TaskRunOutput.
        List of output parameters with name and values.


        :param output_parameters: The output_parameters of this TaskRunOutput.
        :type: list[oci.aidataplatform_dp.models.OutputParameter]
        """
        self._output_parameters = output_parameters

    @property
    def version(self):
        """
        Gets the version of this TaskRunOutput.
        Current version of job run object in repository.


        :return: The version of this TaskRunOutput.
        :rtype: int
        """
        return self._version

    @version.setter
    def version(self, version):
        """
        Sets the version of this TaskRunOutput.
        Current version of job run object in repository.


        :param version: The version of this TaskRunOutput.
        :type: int
        """
        self._version = version

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
